x/ibc/06-solomachine/misbehaviour.go

```diff
@@ -0,0 +1,69 @@
+package solomachine
+
+import (
+	"bytes"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+)
+
+// CheckMisbehaviourAndUpdateState determines whether or not the currently registered
+// public key signed over two different messages with the same sequence. If this is true
+// the client state is updated to a frozen status.
+func CheckMisbehaviourAndUpdateState(
+	clientState clientexported.ClientState,
+	_ clientexported.ConsensusState,
```

Comment:
You need to be storing the consensus state for each sequence correctly, and also using that provided consensus state here to get the right pubkey

Reply:
See above comment, consensus state for solo machines is not stored per-sequence.

Reply:
> See above comment, consensus state for solo machines is not stored per-sequence.

Why not?

Reply:
Because there is no need to, since it can't be used for future verification (unlike commits).